pkg/services/ngalert/api/tooling/definitions/provisioning_templates.go

```diff
@@ -8,23 +8,23 @@ import (
 	"github.com/grafana/grafana/pkg/services/ngalert/models"
 )
 
-// swagger:route GET /api/provisioning/templates provisioning RouteGetTemplates
+// swagger:route GET /api/provisioning/v1/templates provisioning RouteGetTemplates
```

Comment:
**Note: this doesn't block anything about the PR or require changes now, but I want to start a discussion on the topic**

I'd love to figure out how we can generate safer URLs instead of typing these by hand. We're going to write a typo eventually with this approach. Couple that with string constants for things like `contact-points`

If I was generating these in code:
```go
func provisioningAPIPath(version string, pathComponents ...string)
```

Then you can use `path.Join` to construct valid paths. 

Do you think there's a way we can do that with the swagger comments in our definitions files? There's already so much codegen here that I'm loath to add a templating step too, but the manual nature of this makes me nervous.

Reply:
🤔 It might be possible with adding metadata. The extractor we use right now is decent, but it's a little limited in ability to extend it/add custom metadata/use it in a way it doesn't intend. The tool is here: https://github.com/go-swagger/go-swagger 

We already use a set of custom OpenAPI templates. If we can get that extractor to properly read the info we want, that should be enough - the second half is pretty extensible.